### ISSUE TITLE
Data Connectors: Use Environment Variables

### ISSUE THREAD

**Issue by nickthegroot** [hasura/graphql-engine#9777]
### Is your proposal related to a problem?

We use two different Hasura instances depending on whether we're developing locally or deploying publicly, both connected to different databases. We'd like to be able to check in and deploy the same Hasura config, only changing the environment variables depending on the environment.

### Describe the solution you'd like

When defining a data connector (and it's config), allow substituting/specifying environment variables.

For example:

- `backend_configs.yaml`
```yaml
dataconnector:
  weaviate:
      from_env: WEAVIATE_DATABASE_GDC_URL
```

- `databases.yaml`
```yaml
- name: vectors
  ...
  configuration:
    value:
      host:
        from_env: WEAVIATE_DATABASE_URL
```

**Comment by daniel-chambers** [comment]
Thanks for your feedback and suggestions. 🙂

You can set your database URL using environment variables if you use configuration templating. You can specify a `template` property under `configuration`, which takes a [Kriti template](https://hasura.io/docs/latest/api-reference/kriti-templating/) as a string. That template should evaluate to what you'd like your `configuration.value` to be, and you can draw from environment variables.

There's some brief documentation about this on the [Snowflake doco page](https://hasura.io/docs/latest/databases/snowflake/getting-started/cloud/), search for the `Setting the connection string as an environment variable` heading.

As for the backend configs, currently you can't draw from environment variables there; thank you for the suggestion.

As a workaround, you might be able to use the `HASURA_GRAPHQL_METADATA_DEFAULTS` environment variable to configure your data connector agent URL on your local/prod HGE instances, instead of putting it in your YAML files. [Here's an example](https://github.com/hasura/graphql-engine/blob/9a09af4f20452ec934312e7de488abe318d005ae/install-manifests/docker-compose/docker-compose.yaml#L29) of that env var in use.

**Comment by pavan4** [comment]
@daniel-chambers thanks! Currently unable to add `HASURA_GRAPHQL_METADATA_DEFAULTS` env var on Hasura cloud - gives this error message ` failed with error: the HASURA_GRAPHQL_ prefix is reserved by Hasura Cloud.` Any way to get around this?

**Comment by nickthegroot** [comment]
Closing as fixed in #10077 - will go out in next release
